includes/class-smaily-for-cf7-request.php

```diff
@@ -114,15 +114,17 @@ public function get() {
 	 * @return array
 	 */
 	public function post() {
-		$response = array();
-		$auth     = array(
+		$response   = array();
+		$args       = array(
 			'headers' => array(
 				'Authorization' => 'Basic ' . base64_encode( $this->_username . ':' . $this->_password ),
 			),
+			'user-agent' => $this->get_version_info_for_useragent(),
 		);
-		$body = array( 'body' => $this->_data );
-		$args = array_merge( $auth, $body );
-		$subscription_post = wp_remote_post( $this->_url, http_build_query( $args ) );
+		$body       = array( 'body' => $this->_data );
+		$query_data = array_merge( $args, $body );
+
+		$subscription_post = wp_remote_post( $this->_url, http_build_query( $query_data ) );
```

Comment:
Didn't notice this before, but are you sure that `http_build_query` is in the correct place? If I read the documentation correctly then `wp_remote_post` requires second argument to be an array, and `http_build_query` should be used only for `$args['body']` value.

Reply:
I think `WP_Http` actually handles all of it when using `wp_remote_post`.